client/core/account.go

```diff
@@ -214,3 +214,66 @@ func (c *Core) UpdateCert(host string, cert []byte) error {
 
 	return nil
 }
+
+// UpdateDEXHost updates the host for a connection to a dex. The dex at oldHost
+// and newHost must be the same dex, which means that the dex at both hosts use
+// the same public key.
+func (c *Core) UpdateDEXHost(oldHost, newHost string, appPW []byte, certI interface{}) (*Exchange, error) {
+	if oldHost == newHost {
+		return nil, errors.New("old host and new host are the same")
+	}
+
+	crypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return nil, codedError(passwordErr, err)
+	}
+
+	oldDc, _, err := c.dex(oldHost)
+	if err != nil {
+		return nil, err
+	}
+
+	if oldDc.acct.dexPubKey == nil {
+		return nil, fmt.Errorf("cannot update host if dex public key is nil")
+	}
+
+	if oldDc.getPendingFee() != nil {
+		// The notify fee code will have to be completely refactored to allow
+		// this.
+		return nil, fmt.Errorf("host cannot be updated while registration fee is pending")
+	}
+
+	var updatedHost bool
+	newDc, err := c.tempDexConnection(newHost, certI)
+	if newDc != nil { // (re)connect loop may be running even if err != nil
+		defer func() {
+			// Either disconnect or promote this connection.
+			if !updatedHost {
+				newDc.connMaster.Disconnect()
+				return
+			}
+			c.upgradeConnection(newDc)
+		}()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if !newDc.acct.dexPubKey.IsEqual(oldDc.acct.dexPubKey) {
+		return nil, fmt.Errorf("the dex at %s does not have the same public key as %s",
+			oldHost, newHost)
+	}
+
+	err = c.AccountDisable(appPW, oldHost)
```

Comment:
If there are active orders, we'll fail here, but should we check that condition higher up to prevent making an unnecessary connection?